src/processes.c

```diff
@@ -1529,8 +1529,14 @@ static char *ps_get_owner(pid_t pid)
         uid = atoi (line + 5);
         getpwuid_r (uid, &passwd, passwd_buffer, sizeof(passwd_buffer),
                 &passwd_result);
-        if (passwd_result)
+        if (passwd_result) {
             result = sstrdup (passwd_result->pw_name);
+        } else {
+	    // Send the numeric uid if name is not available.
+            char uid_str[6];  // 5 digits plus string terminator.
+            snprintf(uid_str, sizeof(uid_str), "%u", uid);
```

Comment:
With `atoi`, the double conversion is unavoidable. However, `strtoul` allows one to save the end pointer in the string, so you could use something like the following instead:
```
char* uid_end;
uid = strtoul (line + 5, &uid_end, /* base */ 10);
...
} else {
  result = strndup (line + 5, uid_end - line - 5);
}
```
Or we could even add an `sstrndup` function upstream for portability and use that.